kotlinx-collections-immutable/src/main/kotlin/kotlinx/collections/immutable/ImmutableMap.kt

```diff
@@ -27,21 +39,61 @@ public interface ImmutableMap<K, out V>: Map<K, V> {
 }
 
 
-
+/**
+ * A generic persistent collection that holds pairs of objects (keys and values) and supports efficiently retrieving
+ * the value corresponding to each key. Map keys are unique; the map holds only one value for each key.
+ * Modification operations return new instances of the persistent map with the modification applied.
+ *
+ * @param K the type of map keys. The map is invariant on its key type.
+ * @param V the type of map values. The persistent map is covariant on its value type.
+ */
 public interface PersistentMap<K, out V> : ImmutableMap<K, V> {
+    /**
+     * Returns a new instance of the persistent map with the specified [value] associated with the specified [key].
```

Comment:
Many methods that can return this instance are documented as here, i.e. "Returns a **new instance**...".
Do you think it's critical to mention that method can return this instance?

Reply:
It's also possible to write "Returns **an instance** of the persistent..." or "Returns **a** persistent...", without giving any specifics about cases in which new instances are not created.

Reply:
I think it's important to document this aspect of the contract, as it may be used to determine whether the changes were actually made. 

It's ok to shorten the summary to "Returns a persistent ..." or even just describing the operation as if it was mutable, but then we should provide the details in `@return` section:

```
@return a new persistent map instance with the specified [value] associated with the specified [key]; or this instance if no modifications were made in the result of this operation.
```